x/exchange/keeper/fulfillment.go

```diff
@@ -222,35 +222,40 @@ func (k Keeper) FillAsks(ctx sdk.Context, msg *exchange.MsgFillAsksRequest) erro
 }
 
 // SettleOrders attempts to settle all the provided orders.
-func (k Keeper) SettleOrders(ctx sdk.Context, marketID uint32, askOrderIDs, bidOrderIds []uint64, expectPartial bool) error {
+func (k Keeper) SettleOrders(ctx sdk.Context, req *exchange.MsgMarketSettleRequest) error {
```

Comment:
I like how the params were grouped into a single object. However, I recommend using an interface over a message because it is more intuitive for a keeper method, and it will also help down the road with dependency injection. Your proto messages were generated with getters so you should be able to leverage those instead of directly accessing a field.

Reply:
In this instance, I disagree that an interface is more intuitive for a keeper method. First off, though, I got the object for free. It was already defined, already being created and passed in, it has all the data needed, and little (if any) extra. If I made an interface, this function is the only thing that would make use of it, and this `MsgMarketSettleRequest` is the only thing that would implement it. Now think about someone trying to call this method elsewhere. Since it takes in the object, they know, "I need to make one of those objects." But if it took in an interface, now they'd need to either track down which object they can use, or else they have to write their own object.